src/pages/UserSite/HomePage/homePage.tsx

```diff
@@ -23,19 +24,19 @@ export default class HomePage extends React.Component{
             allReviews: []
         }
 
-        this.handleSearchTerm = this.handleSearchTerm.bind(this);
+        this.handleSearchTerm = this.handleSearchTerm.bind(this); // note
         this.handleSubmit = this.handleSubmit.bind(this);
     }
 
     componentDidMount(){
-        const app: any = this;
+        const app = this; // note
```

Comment:
Er det en grunn til at du ikke bare bruker this? Uansett så bør du ikke sette den til type any. Typescript klarer selv å finne ut at når app = this, så er app typen HomePage her.